.github/workflows/ci.yaml

```diff
@@ -1,17 +1,43 @@
 name: CI Pipeline
 
 on:
+  workflow_call:
+    secrets:
+      MAIL_USERNAME:
+        description: 'Username for the email notification step'
+        required: true
+      MAIL_PASSWORD:
+        description: 'Password for the email notification step'
+        required: true
+    inputs:
+      custom_docker_tag:
+        description: 'Custom Docker tag'
+        # If not specified, then the 'latest' tag will be used
+        default: ''
+        required: false
+        type: string
+
   push:
     branches:
       - master
   pull_request:
     branches:
       - master
 
+env:
+  DOCKER_REGISTRY: ghcr.io
+  DOCKER_REPO: angular-react-starter-image
+  DOCKER_TAG: ${{ inputs.custom_docker_tag || 'latest' }}
+  MAIL_SERVER_ADDRESS: smtp.gmail.com
```

Comment:
Let's move `MAIL_SERVER_ADDRESS` and `MAIL_SERVER_PORT` to inputs, cause basically these values could be changed (as well as docker registry, repo). 

All these values could be different, for example today you use Gmail, tomorrow you would like to configure SMTP on AWS SES or Sendgrid. It is much better to allow to pass these variables as inputs for your workflow, instead of changing workfow itself. It is also true for docker registry and repo. Today you use gh, tomorrow you will use AWS ECR or docker registry and would like to change repo name

Reply:
Done